http/httpRequest.h

```diff
@@ -42,18 +42,25 @@ class Request {
     return this->headers_;
   }
 
+	void SetURI(const std::string& uri);
```

Comment:
wait if they are purely virtual then the Request class cannot be instantiated...but we are using the Request class in other places